townhall/myapi/tests.py

```diff
@@ -72,6 +76,21 @@ def test_get_opportunity(self):
         assert opportunity.time == timezone.make_aware(datetime(2024, 7, 20, 10, 0))
         assert opportunity.description == "Deliver food"
         assert opportunity.location == "Vancouver"
+
+    def test_filtered_opportunity_name_and_time(self):
+        start_time = timezone.make_aware(datetime(2024, 7, 20, 0, 0))
+        end_time = timezone.make_aware(datetime(2024, 7, 20, 23, 59))
+        filtered_opportunity_data = townhall_services.FilteredOpportunityData(name="Food", start_time=start_time, end_time=end_time)
+        opportunities = townhall_services.OpportunityServices.filtered_opportunity(filtered_opportunity_data)
+        
+        for o in opportunities:
+            print(f"Opportunity: {o.name}, Time: {o.time}, Location: {o.location}")
+
+        assert len(opportunities) > 0
```

Comment:
when you filter based on Food, you can make a stronger assertion since in `setUp `theres multiple `Opportunity `created so in this test you should expect len to be only 1 since the other one should be filtered away